sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs

```diff
@@ -930,6 +954,35 @@ public virtual void SessionReceiverLinkClosed(string identifier, string sessionI
             }
         }
 
+        [NonEvent]
+        public virtual void SendLinkClosed(
+            string identifier,
+            object sender)
+        {
+            if (IsEnabled())
+            {
+                var link = (SendingAmqpLink)sender;
+                if (link != null)
+                {
+                    Exception exception = link.GetInnerException();
+                    SendLinkClosedCore(
+                        identifier,
+                        exception?.ToString());
+                }
+            }
+        }
+
+        [Event(SendLinkClosedEvent, Level = EventLevel.Informational, Message = "Send Link Closed. Identifier: {0}, linkException: {1}.")]
+        public virtual void SendLinkClosedCore(
+            string identifier,
+            string linkException)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(SendLinkClosedEvent, identifier, linkException);
```

Comment:
So, just a question since I don't know the history. Why do our events send a string representation of the exception instead of the actual exception?

Reply:
Even with the rich data type support added as of NET 4.6, you can't send any old object.
https://docs.microsoft.com/en-us/archive/blogs/vancem/rich-payload-data-in-eventsource-v4-6